### PR TITLE
fix: webkit quicklinks copy for webkit

### PR DESCRIPTION
**Pull request by fschade**
## Description
copying quicklings was broken on webkit based browsers, this is fixed now.

* https://developer.apple.com/forums/thread/691873
* https://webkit.org/blog/10855/

## Related Issue

- Fixeshttps://github.com/owncloud/web/issues/9166

## Motivation and Context
... hmmmm, no bugs, happy dev 🤣
ezgif-2-09128abecb

## How Has This Been Tested?
- unit tests

## Screenshots (if appropriate):
Screenshot 2023-10-19 at 15 25 10

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:
- [X] Code changes
- [X] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-pkg/src/helpers/share/link.ts [hunk 1]
```diff
@@ -22,6 +22,49 @@ export interface CreateQuicklink {
...
+  // the copy action uses the clipboard // clipboardItem api to work around the webkit limitations.
+  //
+  // https://developer.apple.com/forums/thread/691873
+  //
+  // if those apis not available (or like in firefox behind dom.events.asyncClipboard.clipboardItem)
+  // it has a fallback to the vue-use implementation.
+  //
+  // https://webkit.org/blog/10855/
+  const doCopy = async () => {
+    if (typeof ClipboardItem && navigator && navigator.clipboard && navigator.clipboard.write) {
```
Comment (dschmidt) [inline packages/web-pkg/src/helpers/share/link.ts:39]:
```suggestion
    if (typeof ClipboardItem && navigator?.clipboard?.write) {
```

File: packages/web-pkg/src/helpers/share/link.ts [hunk 2]
```diff
@@ -22,6 +22,49 @@ export interface CreateQuicklink {
...
+      await navigator.clipboard.write([
+        new ClipboardItem({
+          'text/plain': createQuicklink(args).then(
+            (link) => new Blob([link.url], { type: 'text/plain' })
+          )
```
Comment (dschmidt) [inline packages/web-pkg/src/helpers/share/link.ts:44]: Does this really accept a promise? seems weird

**Review by dschmidt — COMMENTED** [review]
really nice!

File: packages/web-pkg/src/helpers/share/link.ts (hunk 2 above)
Comment (fschade) [reply packages/web-pkg/src/helpers/share/link.ts:44]: https://developer.mozilla.org/en-US/docs/Web/API/ClipboardItem/ClipboardItem

An Object with the MIME type as the key and data as the value. The data can be represented as a Blob, a String or a Promise which resolves to either a blob or string.

**Review by dschmidt — APPROVED** [review]

**Pull request merged**